lib/open_food_network/order_cycle_permissions.rb

```diff
@@ -154,49 +151,30 @@ def editable_variants_for_incoming_exchanges_from(producer)
       end
     end
 
+    def all_incoming_editable_variants
+      valid_suppliers = visible_enterprises.map do |enterprise|
+        enterprise.id if user_manages_coordinator_or(enterprise)
```

Comment:
The `if` will cause `nil` values in the new array. You can call `compact` on the result or go a different route:

```rb
visible_enterprises.select { |enterprise| user_manages_coordinator_or(enterprise) }.map(&:id)
```

Potentially, we could do it all in SQL but that's probably beyond this PR.